src/cp/constraints/planed_shifts.py

```diff
@@ -0,0 +1,75 @@
+from . import Constraint
+from employee import Employee
+from shift import Shift
+from ..variables import Variable, EmployeeDayShiftVariable
+from ortools.sat.python.cp_model import CpModel
+import logging
+
+
+class PlannedShiftsConstraint(Constraint):
+    KEY = "planned-shifts"
+
+    SHIFT_MAPPING = {
+        "F": 0,
+        "Z": 1,
+        "S": 2,
+        "N": 3,
+        "Z60": 4,
+    }
+
+    EXCLUSIVE_SHIFTS = ["Z60"]  # Shifts that can only be assigned with planned_shifts
+
+    def __init__(self, employees: list[Employee], days: list, shifts: list[Shift]):
+        super().__init__(employees, days, shifts)
+
+    def create(self, model: CpModel, variables: dict[str, Variable]):
+        # Collect all employees who are allowed to have Z60 shifts
+        employees_with_z60 = set()
+
+        for employee in self._employees:
+            if not hasattr(employee, "_planned_shifts") or not employee._planned_shifts:
+                continue
+
+            for day_num, shift_code in employee._planned_shifts:
+                if shift_code in self.EXCLUSIVE_SHIFTS:
+                    employees_with_z60.add(employee.get_key())
+
+                # Existing code für planned shifts...
+                day = next((d for d in self._days if d.day == day_num), None)
+                if not day:
+                    continue
+
+                shift_id = self.SHIFT_MAPPING.get(shift_code)
+                if shift_id is None:
+                    logging.warning(f"Unknown shift code: {shift_code}")
+                    continue
+
+                shift = next((s for s in self._shifts if s.get_id() == shift_id), None)
+                if not shift:
+                    logging.warning(f"Shift with ID {shift_id} not found")
+                    continue
+
+                variable_key = EmployeeDayShiftVariable.get_key(employee, day, shift)
+                if variable_key in variables:
+                    model.add(variables[variable_key] == 1)
```

Comment:
I would prefer if this is built like the vacation days, with `has_planned_shift(day, shift)`